storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -1177,25 +1177,22 @@ bool Redo_Log_Data_Manager::is_error() const { return (error); }
 Redo_Log_Data_Manager::~Redo_Log_Data_Manager() { os_event_destroy(event); }
 
 bool Redo_Log_Data_Manager::stop_at(lsn_t lsn) {
-  bool last_checkpoint = reader.find_last_checkpoint_lsn(&last_checkpoint_lsn);
-  if (last_checkpoint) {
-    msg("xtrabackup: The latest check point (for incremental): '" LSN_PF "'\n",
-        last_checkpoint_lsn);
-  }
   msg("xtrabackup: Stopping log copying thread at LSN " LSN_PF ".\n", lsn);
-
   stop_lsn = lsn;
   os_event_set(event);
   thread.join();
 
   archived_log_monitor.stop();
+  bool last_checkpoint = reader.find_last_checkpoint_lsn(&last_checkpoint_lsn);
+  if (last_checkpoint) {
+    msg("xtrabackup: The checkpoint LSN in redo log file  '" LSN_PF "'\n",
+        last_checkpoint_lsn);
+  }
 
-  scanned_lsn = reader.get_scanned_lsn();
-
-  if (last_checkpoint_lsn > scanned_lsn) {
+  if (lsn > reader.get_scanned_lsn()) {
```

Comment:
Where is scanned_lsn variable set now? We still use it on below check.

Reply:
in case we are using redo log: 
https://github.com/percona/percona-xtrabackup/blob/0a93a67b02fe5a523e54ae1995f5e5bc02822b21/storage/innobase/xtrabackup/src/redo_log.cc#L204

if we are using archive file
https://github.com/percona/percona-xtrabackup/blob/0a93a67b02fe5a523e54ae1995f5e5bc02822b21/storage/innobase/xtrabackup/src/redo_log.cc#L515